### PR TITLE
Frontend#20

### PR DESCRIPTION
**Pull request by Adamk8**
I ended up combining frontend#67 (graphql query for survey page) into this PR too. I've set up the graphql query to grab courses from the mock backend for the survey page. Note that I needed to use the firefox extension for CORS everywhere to make it work along with changing the apollo client port to 4000. The course preferences are now done through combining the 'Able' preference with the 'willing' preference to get a number like below:
image

Heres what the form looks like now:
image

Once we hook up mutation there may be some fields that need to change but this is all the info I had for now

### CODE REVIEW

**Comment by jayden-chan** [comment]
Ah I forgot to mention the `proxy` option in `package.json`. Can you change the GraphQL URL to just `/graphql` and the `proxy` to `http://localhost:4000/` and see if that works? Should also resolve any CORS issues in the browser. This is also how it will need to be set up for the real deployment.

Basically the `proxy` option in `package.json` will redirect any failed routing requests to the URL you put in there (during development only). So the frontend will 404 on `/graphql` and then the proxy will just send that to `localhost:4000` which is where the backend is running.

File: src/components/SurveyCourseList.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+export const SurveyCourseList: React.FC<ChildProps> = (props) => {
+  const { loading, error, data } = useQuery(COURSES, {});
+  const [preferences, setPreferences] = useState<PreferenceListInterface>({});
+
+  useEffect(() => {
+    if (typeof error != "undefined") {
+      console.log(error);
+    }
+  }, [error]);
```
Comment (jayden-chan) [inline src/components/SurveyCourseList.tsx]: Should probably remove debugging stuff before merging

File: src/components/SurveyCourseList.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    } else if (able === "Able" && willing === "Willing") {
+      return 78;
+    } else if (able === "With Effort" && willing === "Very Willing") {
+      return 100;
+    } else if (able === "Able" && willing === "Very Willing") {
+      return 195;
+    } else {
+      return 0;
+    }
+  };
```
Comment (jayden-chan) [inline src/components/SurveyCourseList.tsx]: It's not really a blocker but it's probably a good idea to move some of this data manipulation stuff into it's own file/module

**Review by jayden-chan — COMMENTED** [review]
Looks good, I think we will probably end up wanting the course data in a Zustand store at some point in the future but that should be a pretty easy refactor.

**Review by nicudy — APPROVED** [review]
Looks good!

File: src/components/SurveyCourseList.tsx (hunk 1 above)
Comment (Adamk8) [reply src/components/SurveyCourseList.tsx]: Got rid of this

File: src/components/SurveyCourseList.tsx (hunk 2 above)
Comment (Adamk8) [reply src/components/SurveyCourseList.tsx]: moved to new folder called utils/

**Review by jayden-chan — APPROVED** [review]

**Pull request merged**